### PR TITLE
Dashboards: Split GetDashboardVersions method

### PR DESCRIPTION
**Pull request by idafurjes**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/preference/pref.go [hunk 1]
```diff
@@ -8,6 +8,6 @@ type Service interface {
 	GetWithDefaults(context.Context, *GetPreferenceWithDefaultsQuery) (*Preference, error)
 	Get(context.Context, *GetPreferenceQuery) (*Preference, error)
 	Save(context.Context, *SavePreferenceCommand) error
-	Patch(ctx context.Context, cmd *PatchPreferenceCommand) error
+	Patch(context.Context, *PatchPreferenceCommand) error
```
Comment (mildwonkey) [inline pkg/services/preference/pref.go:11]: 😁 ty these drive me crazy

**Review by mildwonkey — APPROVED** [review]
🎉 looks great!

File: pkg/services/preference/pref.go (hunk 1 above)
Comment (idafurjes) [reply pkg/services/preference/pref.go:11]: haha, I copy paste the signature, and that's how I introduce those :P

**Review by sakjur — APPROVED** [review]
I haven't tested it, the code looks good :+1:

File: pkg/services/sqlstore/store.go [hunk 2]
```diff
@@ -133,4 +133,5 @@ type Store interface {
 	GetDBHealthQuery(ctx context.Context, query *models.GetDBHealthQuery) error
 	SearchOrgs(ctx context.Context, query *models.SearchOrgsQuery) error
 	IsAdminOfTeams(ctx context.Context, query *models.IsAdminOfTeamsQuery) error
+	GetDialect() migrator.Dialect
```
[the remark this answers is not in the document]
Comment (idafurjes) [reply pkg/services/sqlstore/store.go:136]: eeeeh, I guess not

**Pull request merged**